### PR TITLE
COG-1097 User's Role Changed Changelog Record

### PR DESCRIPTION
**Pull request by AlexanderSerbin**
### JIRA Issue Link
https://osi-cwds.atlassian.net/browse/COG-1097

### Technical Description
Published User Role Changed event 

### Tests
- [x] I have included unit tests 
- [x] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

### Types of changes
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by AlexanderSerbin** [comment]
Retest this please

File: src/main/java/gov/ca/cwds/idm/event/AuditEvent.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 
 @JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
-public class AuditEvent<T> {
```
Comment (dmytro-palczewski-osi) [inline src/main/java/gov/ca/cwds/idm/event/AuditEvent.java]: I'm not sure if we need to make class abstract if it does not have abstract methods.
Is not it sufficient for him to be generic?

File: src/main/java/gov/ca/cwds/config/api/idm/Roles.java [hunk 2]
```diff
@@ -53,4 +54,9 @@ private Roles() {}
   public static List<Map<String, String>> findRoles() {
     return Roles.roleList;
   }
+
+  public static String joinRoles(Iterable<String> roles) {
```
Comment (dmytro-palczewski-osi) [inline src/main/java/gov/ca/cwds/config/api/idm/Roles.java]: I'm not sure that it's a good idea to save roles as delimited string

File: src/main/java/gov/ca/cwds/idm/dto/UserChangeLogRecord.java [hunk 3]
```diff
@@ -14,7 +12,7 @@
 
   private String adminRole;
   private String adminName;
-  private List<String> userRoles = new ArrayList<>();
+  private String userRoles;
```
Comment (dmytro-palczewski-osi) [inline src/main/java/gov/ca/cwds/idm/dto/UserChangeLogRecord.java]: I suspect that serialVersionUID should be changed

File: src/main/java/gov/ca/cwds/idm/service/IdmServiceImpl.java [hunk 4]
```diff
@@ -321,10 +329,21 @@ private ExecutionStatus updateUserAttributes(
 
     if (cognitoServiceFacade.updateUserAttributes(userId, existedCognitoUser, updateUserDto)) {
       updateAttributesStatus = SUCCESS;
+      publishUpdateAttributesEvents(existedCognitoUser, updateUserDto);
     }
     return updateAttributesStatus;
   }
 
+  private void publishUpdateAttributesEvents(UserType existedCognitoUser, UserUpdate updateUserDto) {
+    Map<UserAttribute, AttributeType> updatedAttributes =
```
Comment (dmytro-palczewski-osi) [inline src/main/java/gov/ca/cwds/idm/service/IdmServiceImpl.java]: This map was already obtained when command was created. Can we avoid executing the same code twice?

**Comment by AlexanderSerbin** [comment]
Retest this please

File: src/main/java/gov/ca/cwds/idm/dto/UserChangeLogRecord.java (hunk 3 above)
Comment (AlexanderSerbin) [reply src/main/java/gov/ca/cwds/idm/dto/UserChangeLogRecord.java]: Done

**Comment by AlexanderSerbin** [comment]
Retest this please

**Pull request merged**